### PR TITLE
Use MsBuild.exe instead of relying on MsBuild APIs to determine referenced projects

### PR DESCRIPTION
**Pull request by pranavkm**
cc @deepakaravindr @emgarten

### CODE REVIEW

File: src/NuGet.CommandLine/MsBuildUtility.cs [hunk 1]
```diff
@@ -26,48 +27,82 @@ public static bool IsMsBuildBasedProject(string projectFullPath)
...
 
-        public static IEnumerable<string> GetProjectReferences(string projectFullPath)
+        public static IEnumerable<string> GetProjectReferences(string msbuildPath, string projectFullPath)
         {
-            var projectDirectory = Path.GetDirectoryName(projectFullPath);
-            var project = new ProjectInstance(projectFullPath);
+            if (string.IsNullOrEmpty(msbuildPath))
+            {
+                msbuildPath = Path.Combine(
+                    Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86),
```
Comment (emgarten) [inline src/NuGet.CommandLine/MsBuildUtility.cs:35]: Does this work on an X86 machine? If this doesn't exist it should use the non-x86 var.

**Comment by emgarten** [comment]
Check the program files path on an x86 machine. Then :shipit:

File: src/NuGet.CommandLine/MsBuildUtility.cs [hunk 2]
```diff
@@ -26,48 +27,82 @@ public static bool IsMsBuildBasedProject(string projectFullPath)
...
+                    string.Format(
+                        CultureInfo.CurrentCulture,
+                        NuGetResources.MsBuildDoesNotExistAtPath,
+                        msbuildPath));
+            }
 
-            TargetResult targetResult;
-            if (result.ResultsByTarget.TryGetValue(GetProjectReferenceBuildTarget, out targetResult))
+            var targetPath = Path.GetTempFileName();
+            using (var input = typeof(MsBuildUtility).Assembly.GetManifestResourceStream("NuGet.CommandLine.GetProjectsReferencingProjectJsonFiles.target"))
```
Comment (deepakaravindr) [inline src/NuGet.CommandLine/MsBuildUtility.cs:52]: Break the line

File: src/NuGet.CommandLine/GetProjectsReferencingProjectJsonFiles.target [hunk 3]
```diff
@@ -0,0 +1,32 @@
+﻿<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (deepakaravindr) [inline src/NuGet.CommandLine/GetProjectsReferencingProjectJsonFiles.target:1]: please break long lines in the file

File: src/NuGet.CommandLine/MsBuildUtility.cs (hunk 1 above)
Comment (pranavkm) [inline src/NuGet.CommandLine/MsBuildUtility.cs:35]: Yup, it doesn't. Need to switch based on bitness.

**Pull request closed without merging**